### PR TITLE
Created German Cheatsheet Goodie

### PR DESCRIPTION
**Pull request by AdamSC1-ddg**
# Goodie Pull Request Template

Using this template will help us better understand your Instant Answer and assist you when necessary.  Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

**What does your Instant Answer do?** 
Provides a quick reference of beginner phrases for travelers in Germany

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Provides zero-click key phrases for users rather than sifting through translation cites

**What is the data source for your Instant Answer? (Provide a link if possible)**
N/A

**Why did you choose this data source?**
N/A

**Are there any other alternative (better) data sources?**
N/A

**What are some example queries that trigger this Instant Answer?**
"German Cheatsheet" "English to German Cheatsheet"

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
travelers 

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No

**Which existing Instant Answers will this one supercede/overlap with?**
N/A

**Are you having any problems? Do you need our help with anything?**
N/A

**Where did you hear about DuckDuckHack? (For first time contributors)**
DDG team

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
http://i.imgur.com/1lw8Ad1.png

## Checklist
Please place an 'X' where appropriate.

```
- [x] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [x] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [x] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [x] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [x] Google Chrome
        - [x] Firefox
        - [] Opera
        - [x] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

### CODE REVIEW

File: t/GermanCheatsheet.t [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+zci answer_type => "german_cheatsheet";
+zci is_cached   => 1;
+
+ddg_goodie_test(
+    [qw( DDG::Goodie::GermanCheatsheet )],
+    # At a minimum, be sure to include tests for all:
+    # - primary_example_queries
+    # - secondary_example_queries
+    'example query' => test_zci('query'),
```
Comment (mintsoft) [inline t/GermanCheatsheet.t]: Can you add tests here? At the very minimum for the `primary_example_queries "German Cheatsheet", "English to German Cheatsheet";` 

There's a section in the documentation (https://duck.co/duckduckhack/goodie_testing_triggers) however just comment if you have any issues and someone'll swing in to help :)

File: share/goodie/german_cheatsheet/german_cheatsheet.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
+<div class="gimp-container">
+
+    <div class="gimp-column">
+        <b><u>Basics</u></b>
+        <table class="gimp-table">
+            <tr><td><b>Welcome</b></td><td>Willkommen</td></tr>
```
Comment (mintsoft) [inline share/goodie/german_cheatsheet/german_cheatsheet.html]: I'd probably put a class on the first `<td>` and do the emboldening in css

File: share/goodie/german_cheatsheet/german_cheatsheet.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            <tr><td><b>Hello</b></td><td>Hallo/Guten Tag</td></tr>
+            <tr><td><b>How are you?</b></td><td>Wie gehts?</td></tr>
+            <tr><td><b>I'm fine thanks!</b></td><td>Mir geht es gut, danke.</td></tr>
+            <tr><td><b>What's your name?</b></td><td>Wie Heisst du?</td></tr>
+            <tr><td><b>My name is...</b></td><td>Ich heisse...</td></tr>
+        </table>
+
+        <b><u>Getting Help</u></b>
```
Comment (mintsoft) [inline share/goodie/german_cheatsheet/german_cheatsheet.html]: Symantically I think these should probably an `<h4>` or similiar?

File: lib/DDG/Goodie/GermanCheatsheet.pm [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            twitter => "adamscochran";
+
+# Triggers
+triggers any => "german cheatsheet", "cheatsheet german";
+
+my $HTML = share("german_cheatsheet.html")->slurp(iomode => "<:encoding(UTF-8)");
+
+
+handle remainder => sub {
+    return
```
Comment (mintsoft) [inline lib/DDG/Goodie/GermanCheatsheet.pm]: Currently this will return a response for queries like `buy german cheatsheet book`. It's probably worth guarding the query with something like:
```perl
handle remainder => sub {
    return if $_;
    return heading -> "English to German Cheat Sheet",
        html => $HTML;
};

File: lib/DDG/Goodie/GermanCheatsheet.pm [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        heading => "English to German Cheat Sheet",
```
Comment (mintsoft) [inline lib/DDG/Goodie/GermanCheatsheet.pm]: Can you add `text` output too?

**Comment by mintsoft** [comment]
@AdamSC1-ddg Thanks for the contribution! I've added some direct feedback against the diff

File: share/goodie/german_cheatsheet/german_cheatsheet.html (hunk 2 above)
Comment (AdamSC1-ddg) [inline share/goodie/german_cheatsheet/german_cheatsheet.html]: I'm not sure what this means to be honest!

File: share/goodie/german_cheatsheet/german_cheatsheet.html (hunk 3 above)
Comment (AdamSC1-ddg) [inline share/goodie/german_cheatsheet/german_cheatsheet.html]: For the sub-section titles?

File: lib/DDG/Goodie/GermanCheatsheet.pm (hunk 4 above)
Comment (AdamSC1-ddg) [inline lib/DDG/Goodie/GermanCheatsheet.pm]: Modified this

File: share/goodie/german_cheatsheet/german_cheatsheet.html (hunk 2 above)
Comment (mintsoft) [inline share/goodie/german_cheatsheet/german_cheatsheet.html]: Sorry! I meant something like:
```html
<table class="gimp-table">
    <tr><td class='english'>Welcome</td><td>Willkommen</td></tr>
    <tr><td class='english'>Hello</td><td>Hallo/Guten Tag</td></tr>
```
Then in the css do:
```css
td.english {
   font-weight: bold;
}
```

**Comment by zachthompson** [comment]
Replaced by https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1287
